kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/steps/KafkaSteps.java

```diff
@@ -41,14 +45,15 @@ public static void createTopic(String deployNamespace, String topicName, String
         kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
                 .withImage(Constants.TEST_CLIENTS_IMAGE)
                 .withName(podName)
-                .withRestartPolicy("Never")
-                .withCommand("admin-client")
-                .withArgs("topic", "create", "--bootstrap-server=" + bootstrap, "--topic=" + topicName, "--topic-partitions=" + partitions,
+                .withRestartPolicy(NEVER_POLICY)
+                .withCommand(ADMIN_CLIENT_COMMAND)
+                .withArgs(TOPIC_COMMAND, "create", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName, "--topic-partitions=" + partitions,
                         "--topic-rep-factor=" + replicas)
                 .done();
 
-        DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofSeconds(30));
-        LOGGER.debug("Admin client pod log: {}", kubeClient().logsInSpecificNamespace(deployNamespace, podName));
+        DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofMinutes(1));
```

Comment:
It seems odd to give this container 1 minute, and the others get 30 seconds.  We know that we do that because that's how we happen to use the this API _at the moment_ (create first, delete second...).   Classes shouldn't need to know how they get used.  That makes for fragile code. 

As a minimum I would extract the timeout as a constant and give them all 1 minute.

A slightly better solution would be to apply a different timeout whilst the Pod is in the Pending phase.  Only start applying the timeout passed by the caller once the pod has gone beyond Pending.
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase

something like this (untested)

```java
        // Wait until the pod has a non-null phase, and that phase and is something other than Pending.
        with()
                 .alias("await pod to leave pending phase")
                 .timeout(Duration.ofMinutes(1))
                .until(() -> Optional.of(kubeClient().getPod(namespaceName, podName)).map(Pod::getStatus),
                        s -> s.map(PodStatus::getPhase).filter(Predicate.not("pending"::equalsIgnoreCase)).isPresent());
```

Reply:
I agree giving all a timeout of 1 minute, but waiting for leaving `pending` phase seems not to be right to me. What if leaving `pending` phase means that we end on `failed` or `error`? We should not continue

Reply:
I was suggesting a two-stage approach.

1. apply a timeout waiting for the pod to leave the pending phase.  Make it a generous timeout so that it copes with a slow download from a container registry.
2. apply the logic to wait for the pod to reach a terminal condition, or timeout if it stays running for too long.

2. is pretty much what you have now.